api/tools/src/main/java/org/pmiops/workbench/tools/CreateWgsCohortExtractionBillingProjectWorkspace.java

```diff
@@ -38,38 +47,81 @@
 
   private static Options options =
       new Options()
-          .addOption(billingAccountOpt)
+          .addOption(configJsonOpt)
           .addOption(billingProjectNameOpt)
           .addOption(workspaceNameOpt)
           .addOption(ownersOpt);
 
   private static final Logger log =
-      Logger.getLogger(CreateExtractionBillingProjectWorkspace.class.getName());
+      Logger.getLogger(CreateWgsCohortExtractionBillingProjectWorkspace.class.getName());
 
-  @Bean
-  ImpersonatedServiceAccountApiClientFactory apiClientFactory(WorkbenchConfig config) {
-    // TODO : replace string with `config.auth.extractionServiceAccount` once that value is in the
-    // test db
+  WorkbenchConfig workbenchConfig(String configJsonFilepath) throws IOException {
+    ObjectMapper jackson = new ObjectMapper();
+    String rawJson =
+        new String(Files.readAllBytes(Paths.get(configJsonFilepath)), Charset.defaultCharset());
+
+    String strippedJson = rawJson.replaceAll("\\s*//.*", "");
+    JsonNode newJson = jackson.readTree(strippedJson);
+
+    return (new Gson()).fromJson(newJson.toString(), WorkbenchConfig.class);
+  }
+
+  ImpersonatedServiceAccountApiClientFactory wgsCohortExtractionServiceAccountApiClientFactory(
+      WorkbenchConfig config) throws IOException {
     return new ImpersonatedServiceAccountApiClientFactory(
-        "wgs-cohort-extraction@all-of-us-workbench-test.iam.gserviceaccount.com", config);
+        config.wgsCohortExtraction.serviceAccount, config.firecloud.baseUrl);
+  }
+
+  private String getExtractionPetSa(String workspaceNamespace, WorkbenchConfig workbenchConfig)
+      throws IOException, InterruptedException {
+    String accessToken =
+        ImpersonatedServiceAccountApiClientFactory.getAccessToken(
+            workbenchConfig.wgsCohortExtraction.serviceAccount);
+    log.info("Extraction SA Access Token: " + accessToken);
+
+    Request request =
+        new Request.Builder()
+            .url(
+                workbenchConfig.firecloud.samBaseUrl
+                    + "/api/google/v1/user/petServiceAccount/"
+                    + workspaceNamespace)
+            .addHeader("Authorization", "Bearer " + accessToken)
+            .build();
+
+    final OkHttpClient client = new OkHttpClient();
+    Response response = client.newCall(request).execute();
+
+    // The first call seems to always fail for some reason. Just retry a few times
+    int retries = 0;
+    while (!response.isSuccessful() && retries < 3) {
```

Comment:
Shouldn't this log an error if it fails after 3 times? looks like this will just result in an indirect JSON parsing error downstream